internal/pkg/cli/svc_init.go

```diff
@@ -124,16 +126,17 @@ func newInitSvcOpts(vars initSvcVars) (*initSvcOpts, error) {
 		Prog:     termprogress.NewSpinner(log.DiagnosticWriter),
 		Deployer: cloudformation.New(sess),
 	}
+	fs := &afero.Afero{Fs: afero.NewOsFs()}
 	return &initSvcOpts{
 		initSvcVars: vars,
 
-		fs:                    &afero.Afero{Fs: afero.NewOsFs()},
+		fs:                    fs,
 		init:                  initSvc,
 		prompt:                prompter,
 		sel:                   sel,
 		dockerEngineValidator: exec.NewDockerCommand(),
-		setupParser: func(o *initSvcOpts) {
-			o.df = exec.NewDockerfile(o.fs, o.dockerfilePath)
+		initParser: func(s string) dockerfileParser {
+			return exec.NewDockerfile(fs, s)
```

Comment:
What do you think of:
```suggestion
		opts := &initSvcOpts{...}
		opts.dockerfile = func(path string) dockerfileParse {
		     if opts.df != nil { return opts.df }
		     opts.df = exec.NewDockerfile(o.fs, o.dockerfilePath)
		     return opts.df
		}
```